src/SamplesApp/SamplesApp.UITests/RuntimeTests.cs

```diff
@@ -38,8 +39,30 @@ IAppQuery AllQuery(IAppQuery query)
 			var runTestCount = new QueryEx(q => AllQuery(q).Marked("runTestCount"));
 			var unitTestsControl = new QueryEx(q => AllQuery(q).Marked("UnitTestsRootControl"));
 
-			bool IsTestExecutionDone()
-				=> runningState.GetDependencyPropertyValue("Text")?.ToString().Equals("Finished", StringComparison.OrdinalIgnoreCase) ?? false;
+			async Task<bool> IsTestExecutionDone()
+			{
+				var sw = Stopwatch.StartNew();
+				Exception lastException = null;
+				do
+				{
+					try
+					{
+						return runningState.GetDependencyPropertyValue("Text")?.ToString().Equals("Finished", StringComparison.OrdinalIgnoreCase) ?? false;
+					}
+					catch (Exception e)
+					{
+						lastException = e;
```

Comment:
Do you understand when this typically fails? Is it related to the full-screen tests?

Reply:
Yes, I'll be changing the doc now to explain that